### ISSUE TITLE
Replacing e-mail?

### ISSUE THREAD

**Issue by fabiosantoscode** [dominictarr/cyphernet#18]
I imagine that this protocol would be eligible for creating an e-mail replacement.

E-mail is quite asking for new features which you get for free under this protocol:
- Native (without GPG) authentication of the sender, and encryption of the message
- Authentication of the whole conversation. If you are replying to an e-mail, there would be a link to the previous message, and there would be no reason not to have one. 
- No problem accessing messages on untrusted networks
- No passwords, just a private key you can keep around

As I understand it, this protocol allows an e-mail to refer to previous messages on a conversation without copying the text, which has the hassle of the previous message being stored onto the current one (possibly modified!).

"addresses" could be SHA hashes of pubkeys, so if you are paranoid you can exchange public keys and be sure you are always talking to that person.

Would this protocol be feasible to create a new way to send electronic messages?

**Comment by dominictarr** [comment]
yes! except that this doesn't address the problem of actually routing the messages...
since having this idea, I've decided to backpedal, and focus on something simpler that is address the routing problem better and is more like twitter (i.e. each user produces a feed (of arbitrary data) that can be followed by others) except that messages can be passed on by your followers - you do not have to get a message directly from a whoever you follow - getting it from another node who follows a given node also works.

I've been catching up on other stuff I've already started but not finished and havn't published anything about this idea yet though... If you want to talk about this stuff come hang out in #stackvm on freenode!
